### ISSUE TITLE
Maemo Leste developing workflow documentation is missing

### ISSUE THREAD

**Issue by spinal84** [maemo-leste/bugtracker#240]
We should have workflow documented somewhere, so Maemo Leste developers can work and interact effectively.
See bug https://github.com/maemo-leste/bugtracker/issues/234.

**Comment by spinal84** [comment]
I propose this generic workflow:
1. Upload your changes to maemo/ascii-devel branch
2. Build the package in Jenkins
3. Test your changes for a while (about 2 weeks should be fine)
4. Create merge request to master branch
5. Other people will check what you've done, discuss your work and propose how to make it better
6. If everything is ok, your work is merged to master branch
7. From master branch it goes to maemo/ascii branch
8. Jenkins build is scheduled for stable repo.

In short: "master" branch follows "maemo/ascii-devel" and "maemo/ascii" follows "master".

**Comment by spinal84** [comment]
@freemangordon, @MerlijnWajer, @parazyd, any thoughts, suggestions?

**Comment by MerlijnWajer** [comment]
Few things. I am personally not too happy having to use Github's Pull Request feature for everything I do. The other thing is that I'm not sure if it makes sense to always go the ascii-devel path. Some changes are just incredibly trivial and do not warrant "two weeks" of testing, like this: https://github.com/maemo-leste/ke-recv/commit/b1a7b3674ec2f9605ada7bb0f23d1d99c53d9c59

Maybe when we have a beta/stable release we can "enforce" a path that requires always using `-devel` first. That said, I am fine with the branch-flow as proposed otherwise.

**Comment by spinal84** [comment]
So, should we just stay on what we already have, unchanged?

**Comment by freemangordon** [comment]
In general, I am fine with the proposed workflow, BUT, after we release the beta :). Until then I'd recommend something like:
- For those who have push rights - up to them to decide which branch (ascii-devel or master) to push to.
- If a pull request (from a 3rd-party dev) is merged - this should be done in ascii-devel and the proposed workflow should be used, but with reduced quarantine period (3 days lets say)

@spinal84: ATM we just don't have the resources needed to do a proper QA. devel->testing->stable is a very good thing, however, I have very bad experience with it with maemo repos. Packages were lying there in -devel and -testing for years, just because there were no enough testers to promote. Until we have a wider user base, it just doesn't make sense.

**Comment by MerlijnWajer** [comment]
Maybe we should make explicit what the workflow until beta is then. I would say:

* maemo/ascii-devel - for your test builds, new dh versions, huge changes you want tested
* maemo/ascii - go here right away for smaller changes or changes to things that are considered `experimental` anyway.
* master - either equal to maemo/ascii or maemo/ascii-devel
* maemo/beowulf - for beowulf porting

**Comment by MerlijnWajer** [comment]
I think this complements what @freemangordon said above.

**Comment by MerlijnWajer** [comment]
I've taken a stab at this, here: https://leste.maemo.org/Development#Guides

**Comment by MerlijnWajer** [comment]
I think once the beowulf port if done, we should give this some more priority, and also figure out how we help people port applications, especially with regard to https://github.com/maemo-leste-extras
